docs/design/plugin-registry.md

```diff
@@ -0,0 +1,131 @@
+---
+title: plugin-registry
+authors:
+  - @kaovilai
+reviewers:
+  - @shubham-pampattiwar
+  - @eemcmullan
+  - @savitharaghunathan
+  - @dymurray
+  - @shawn-hurley
+approvers:
+  - @dymurray
+  - @shawn-hurley
+  - @shubham-pampattiwar
+creation-date: 2022-06-14
+last-updated: 2022-06-14
+status: implementable
+---
+
+# Plugin Registry
+
+Plugin registry enhancement replaces current usage of [registry](https://github.com/konveyor/distribution/) [deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) insecure [routes](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/networking/configuring-routes) by talking to registry API server inside the velero container itself via openshift-velero-plugin.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Open Questions [optional]
+
+Do we want registry logs to be included in velero backup and/or restore logs?
```

Comment:
No I think there are 2 things we are talking about here. 
1. the plugin logs that records the process of backing up and restoring images.
2. The registry library doing the actual push/pull of the images.

I think 1. should be in the backup/restore for sure. 2. feels overly verbose

Reply:
Yes. Plugin logs. However the logs generated from registry will not be in backup logs in object storage since the velero backup logger isn't being passed to the ServeHTTP method.

Reply:
We do want them but I believe that per current implementation it is not possible.

Reply:
On the positive side, this means that we can leave log level as debug for the registry component without additional configuration flags.